### PR TITLE
Connect Study Plan Service to Azure Database for MySQL

### LINKED ISSUE #9: [DEV] POC - Setup Azure Database for MySQL Instance
Read about Azure Database for MySQL and create a Azure Database for MySQL on Microsoft Azure Cloud.

### LINKED ISSUE #10: [DEV] Connect to Azure Database for MySQL and Create a New User
Once the Azure Database for MySQL Instance is up and running, connect to Azure Database for MySQL and Create a New User with required permissions.

### LINKED ISSUE #11: [DEV] POC - Password-less Authentication to Azure Database for MySQL using Managed Identities.
- A User Assigned Managed Identity is created and is used for MySQL authentication
- Setup should work in local - Grant proper Role to user account.

### LINKED ISSUE #12: [DEV] Implementation - Connect to Azure Database for MySQL from Study Plan Service
- Add the required DB dependencies
- Update the application-.yaml file with the required DB configurations.
- Create the DataSource bean configurations for connecting to Azure Database for MySQL using proper Qualifiers

### PR DESCRIPTION
**Pull request by adityashukla11**
- Added Profiles for aws / azure.
closes #11 
closes #10 
closes #12

### CODE REVIEW

File: src/main/resources/application-azure.properties [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#Logging
+logging.level.org.hibernate.SQL=DEBUG
+
+#AzureConfiguration
+spring.datasource.azure.passwordless-enabled = true
+
+#Datasource Configurations
+nfyc.azure.datasource.url=
+nfyc.azure.datasource.username=
```
Comment (roopa1099) [inline src/main/resources/application-azure.properties:15]: Why is it kept blank?

File: src/main/java/com/nfyc/studyplanservice/config/cache/database/AzureMySQLConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package com.nfyc.studyplanservice.config.cache.database;
+
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.*;
```
Comment (roopa1099) [inline src/main/java/com/nfyc/studyplanservice/config/cache/database/AzureMySQLConfiguration.java]: Import only the required part.

File: src/main/resources/application-azure.properties (hunk 1 above)
Comment (adityashukla11) [reply src/main/resources/application-azure.properties:15]: url may vary as i might delete the mysql instance and create again

**Comment by adityashukla11** [comment]
Reviews Incorporated  @roopa1099

**Review by roopa1099 — APPROVED** [review]

**Pull request merged**